s4e-web/src/app/state/session/session.service.ts

```diff
@@ -1,97 +1,102 @@
-    if (token == null) {
-      localStorage.removeItem('token');
-    } else {
-      localStorage.setItem('token', token);
-    }
+  getProfile$(): Observable<Session | null> {
+    const r = this.http.get<Session>(
+      `${this.CONFIG.apiPrefixV1}/users/me`,
+      {
+        headers: {[ERROR_INTERCEPTOR_SKIP_HEADER]: HTTP_401_UNAUTHORIZED.toString()}
+      }).pipe(
+      tap(profile => this.store.update({...profile})),
+      catchError(() => of(null)),
+      shareReplay(1)
+    );
+
+    r.subscribe();
+
+    return r;
+  }
 
-    if (email == null) {
-      localStorage.removeItem('email');
-    } else {
-      localStorage.setItem('email', email);
-    }
+  resetPassword(oldPassword: string, newPassword: string) {
+    const url = '/api/v1/password-change';
+    return httpPostRequest$(this.http, url, {oldPassword, newPassword}, this.store)
+      .pipe(tap(() => this._notificationService.addGeneral({
+        type: 'success',
+        content: 'Hasło zostało zmienione'
+      })));
   }
 
   @action('login')
   login(formState: LoginFormState) {
-    this.sessionStore.setLoading(true);
+    this.store.setLoading(true);
     const {login: email, password, ...x} = formState;
-    const url = `${this.CONFIG.apiPrefixV1}/token`;
-    this.http.post<LoginRequestResponse>(url, {email, password},
-      {headers: {[ERROR_INTERCEPTOR_CODES_TO_SKIP]: `${HTTP_404_BAD_REQUEST},${HTTP_401_UNAUTHORIZED}`} })
+    const url = `${this.CONFIG.apiPrefixV1}/login`;
+    this.http.post(url, {email, password},
+      {headers: {[ERROR_INTERCEPTOR_CODES_TO_SKIP]: `${HTTP_404_BAD_REQUEST},${HTTP_401_UNAUTHORIZED}`}})
       .pipe(
-        catchErrorAndHandleStore(this.sessionStore),
-        tap(data => this.setToken(data.token, data.email)),
-        switchMap(data => this.profileService.get$()),
-        finalize(() => this.sessionStore.setLoading(false))
+        catchErrorAndHandleStore(this.store),
+        switchMap(data => this.getProfile$()),
+        tap(() => this.store.update({email: formState.login})),
+        finalize(() => this.store.setLoading(false))
       )
-      .subscribe(data => {
-        this._navigateToApplication();
-      });
+      .subscribe(data => this._navigateToApplication());
   }
 
   @action('logout')
   logout() {
-    if (this.sessionQuery.isLoggedIn()) {
-      this.setToken(null, null);
-      resetStores({
-        exclude: ['Notification']
+    const router: Router = InjectorModule.Injector.get(Router);
+    this.http.post(`${this.CONFIG.apiPrefixV1}/logout`, {})
+      .subscribe(() => {
+        this.store.reset();
+        router.navigate(['/']);
       });
-      this.router.navigateByUrl('login');
-    }
+  }
+
+  clearError() {
+    this.store.setError(null);
   }
 
   private _navigateToApplication(): void {
+    const router: Router = InjectorModule.Injector.get(Router);
+
     !!this._backLink && this._backLink !== ''
-      ? this.router.navigateByUrl(this._backLink)
```

Comment:
Why haven't you use context `this.router`?

Reply:
Because of circular dependency. This is a small hack, which probably should be refactored as part of separate PR. (separate `S4EConfig` from config loader should break circular dependency